src/LevelZeroIOGroup.hpp

```diff
@@ -0,0 +1,122 @@
+#define LEVELZEROIOGROUP_HPP_INCLUDE
+
+#include <map>
+#include <vector>
+#include <string>
+#include <memory>
+#include <functional>
+
+#include "IOGroup.hpp"
+#include "LevelZeroSignal.hpp"
+
+namespace geopm
+{
+    class PlatformTopo;
+    class LevelZeroDevicePool;
+
+    /// @brief IOGroup that provides signals and controls for Accelerators
+    class LevelZeroIOGroup : public IOGroup
+    {
+        public:
+            LevelZeroIOGroup();
+            LevelZeroIOGroup(const PlatformTopo &platform_topo, const LevelZeroDevicePool &device_pool);
+            virtual ~LevelZeroIOGroup() = default;
+            std::set<std::string> signal_names(void) const override;
+            std::set<std::string> control_names(void) const override;
+            bool is_valid_signal(const std::string &signal_name) const override;
+            bool is_valid_control(const std::string &control_name) const override;
+            int signal_domain_type(const std::string &signal_name) const override;
+            int control_domain_type(const std::string &control_name) const override;
+            int push_signal(const std::string &signal_name, int domain_type, int domain_idx)  override;
+            int push_control(const std::string &control_name, int domain_type, int domain_idx) override;
+            void read_batch(void) override;
+            void write_batch(void) override;
+            double sample(int batch_idx) override;
+            void adjust(int batch_idx, double setting) override;
+            double read_signal(const std::string &signal_name, int domain_type, int domain_idx) override;
+            void write_control(const std::string &control_name, int domain_type, int domain_idx, double setting) override;
+            void save_control(void) override;
+            void restore_control(void) override;
+            std::function<double(const std::vector<double> &)> agg_function(const std::string &signal_name) const override;
+            std::function<std::string(double)> format_function(const std::string &signal_name) const override;
+            std::string signal_description(const std::string &signal_name) const override;
+            std::string control_description(const std::string &control_name) const override;
+            int signal_behavior(const std::string &signal_name) const override;
+            static std::string plugin_name(void);
+            static std::unique_ptr<IOGroup> make_plugin(void);
+        private:
+            void init(void);
+
+            void register_derivative_signals(void);
+            void register_signal_alias(const std::string &alias_name, const std::string &signal_name);
+            void register_control_alias(const std::string &alias_name, const std::string &control_name);
+
+            const PlatformTopo &m_platform_topo;
+            const LevelZeroDevicePool &m_levelzero_device_pool;
+            bool m_is_batch_read;
+            std::vector<uint64_t> m_initial_power_limit;
```

Comment:
Currently unused.  Remove until needed for save/restore.

Reply:
Expanded to include signals we expect to need for the restore portion of save restore.